sched/group/group.h

```diff
@@ -80,6 +69,7 @@ int  group_bind(FAR struct pthread_tcb_s *tcb);
 int  group_join(FAR struct pthread_tcb_s *tcb);
 #endif
 void group_leave(FAR struct tcb_s *tcb);
+void group_drop(FAR struct task_group_s *group);
```

Comment:
why we have both group_drop and group_addrenv_drop

Reply:
Because the address environment is needed even if the group is destroyed. The kernel uses mappings from user processes i.e. the kernel page directory is copied to each user process, and when the kernel is running it "lends" the mappings from the user process. Sometimes this is called "lazy tlb", the reason is that no TLB flush is needed when switching from user -> kernel.

Reply:
Also, as kernel threads to not have an address environment of their own, mm_group is NULL for them.